### PR TITLE
fix(tracker): support configurable Type fields

### PR DESCRIPTION
**Pull request by lhpaul**
## Summary

- add configurable GitHub Projects classification field support via `issue_tracker.custom_fields.type_field`
- keep `Type` as the default while falling back to `Custom Type` and `CustomType` for targeted reads and metadata lookup
- add `--type` to `add-backlog-item.sh create` so new backlog items can set classification alongside Priority and Size
- document the config key and update regression coverage

Closes #1144

## Verification

- `bash scripts/development-workflow/tests/test-workflow-lib-github-projects.sh`
- `bash scripts/development-workflow/tests/test-add-backlog-item.sh`
- `bash -n scripts/development-workflow/workflow-lib.sh scripts/development-workflow/add-backlog-item.sh scripts/development-workflow/tests/test-workflow-lib-github-projects.sh scripts/development-workflow/tests/test-add-backlog-item.sh`
- `shellcheck --severity=warning -x scripts/development-workflow/add-backlog-item.sh scripts/development-workflow/workflow-lib.sh scripts/development-workflow/tests/test-add-backlog-item.sh scripts/development-workflow/tests/test-workflow-lib-github-projects.sh`
- `python3 scripts/lint/workflow-shell-guard-lint.py --base-ref origin/develop-external-review-loop-hardening`
- `npx markdownlint-cli2 "docs/workflow/development-workflow/integrations/github-projects.md" "docs/workflow/development-workflow/protocols/00-add-backlog-item-protocol.md" "CHANGELOG.md"`
- `find docs/specs/developments docs/testing/workflow -name "*.md" -print0 | xargs -0 python3 scripts/lint/markdown-heuristic-lint.py CHANGELOG.md`
- `bash scripts/lint/check-changelog-duplicate-headers.sh CHANGELOG.md`
- `git diff --check`

## Live Tracker Checks

- Created #1144 through the updated local helper with `--type Workflow --priority Medium --size S`.
- Direct GraphQL readback confirmed #1144 has `Status=In Development`, `Type=Workflow`, `Priority=Medium`, and `Size=S`.
- Backfilled #1112 and #1114-#1118 to `Type=Workflow`; direct GraphQL readback confirmed #1112-#1118 now all have `Type=Workflow`.

## Test Harness Coverage Checklist

- Empty / zero-length input: existing empty project ID and missing project number tests cover empty helper inputs.
- Whitespace-only input: not applicable; this change passes fixed option values and field names, and the tested CLI rejects missing values before field mutation.
- Boundary values: covered by absent config, configured `Custom Type`, fallback-only `Custom Type`, and default `Type` cases.
- Missing / absent environment variables: existing missing project number/provider guard tests continue to cover fail-open behavior.
- Concurrent / parallel execution: not applicable; helper state is per-process cache only and no shared filesystem state is written.
- Negative assertions: covered by no-mutation checks for missing fields/options/provider/project and by no default Type update in `add-backlog-item.sh create`.
- Single EXIT trap: verified one cleanup trap in each touched test harness.
- Variable-length quoting safety: `Custom Type` with a space is covered in read/update lookup tests.
- BASH_SOURCE / HARNESS_MODE guard placement: not applicable; these harnesses execute directly and source only `workflow-lib.sh`.
- Sourced-function ordering: verified test overrides are defined after sourcing `workflow-lib.sh` where intentional.

## Script-Accuracy Self-Check

Scripts: `scripts/development-workflow/add-backlog-item.sh`, `scripts/development-workflow/workflow-lib.sh`

- `--type` flag and usage text: verified in `add-backlog-item.sh`.
- `--type` calls `update_tracker_type_best_effort`: verified in `add-backlog-item.sh`.
- `type_field` custom field lookup: verified in `workflow-lib.sh`.
- Fallback names `Type`, `Custom Type`, `CustomType`: verified in `workflow-lib.sh`.
- `list_open_workflow_type_issues` reads configured/custom field values: verified in `workflow-lib.sh`.

## Pre-Submission Self-Review

Pre-submission self-review: stale markers none, caller consistency verified across helpers/docs/tests, coverage addresses the diagnosed #1144 problem and proposed fix.

---

> [!NOTE]
> **Low Risk**
> Workflow-script and documentation changes with fail-open best-effort updates; no auth or production runtime paths. Misconfigured field names would only skip or warn on classification updates.
> 
> **Overview**
> GitHub Projects boards that name their classification column something other than **`Type`** can now be wired through **`issue_tracker.custom_fields.type_field`** (for example **`Custom Type`**). Tracker helpers use that name for reads and writes, default to **`Type`**, and fall back to **`Custom Type`** / **`CustomType`** when resolving values and field metadata.
> 
> **`add-backlog-item.sh create`** gains **`--type`** (`Feature`, `Bug`, `Refactor`, `Workflow`) so new issues can set the project classification field alongside Priority and Size. **`workflow-lib.sh`** updates targeted GraphQL reads, type-field lookup/caching, **`update_tracker_type_best_effort`**, and **`list_open_workflow_type_issues`** so configured custom fields win over stale **`type`** values on board scans.
> 
> Docs (`.ai-dev-workflow.yaml` example, **`github-projects.md`**, backlog protocol) and regression tests cover the new flag and custom-field behavior.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 308581c038d835bae24a3831ae464d293a8eaa64. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by lhpaul** [comment]
bugbot run

**Comment by lhpaul** [comment]
### Automated Reviewer Loop Summary

**Result:** clean — no blocking findings
**Platforms:** pr-agent (clean), bugbot (clean)
**Findings:** 0 blocking, 0 suggestions

**Ready reviewer phase:** No net-new blocker was found after the draft GitHub gate.
**Ready-phase platforms:** bugbot

**Advisory findings (non-blocking):**
- pr-agent: Possible Issue - Possible Issue
  Source: [view comment](https://github.com/lhpaul/ai-dev-framework-template/pull/1145#issuecomment-4870182248)
- pr-agent: Possible Issue - Possible Issue
  Source: [view comment](https://github.com/lhpaul/ai-dev-framework-template/pull/1145#issuecomment-4870182248)

*Posted automatically by `pr-review-loop.sh`.*

File: scripts/development-workflow/workflow-lib.sh [hunk 1]
```diff
@@ -1691,17 +1719,23 @@ workflow_github_project_type_field_json() {
...
+candidate_names = []
+for name in (configured, 'Type', 'Custom Type', 'CustomType'):
+    if name and name not in candidate_names:
+        candidate_names.append(name)
 for field in fields:
-    if field.get('name') == 'Type':
+    if field.get('name') in candidate_names:
         field_json = json.dumps({
             'field_id': field.get('id') or '',
+            'field_name': field.get('name') or '',
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply scripts/development-workflow/workflow-lib.sh]: Fixed in 308581c by making type field metadata lookup iterate candidate names in priority order, so configured `type_field` wins before fallback fields. Added `configured_custom_type_beats_type_field_*` regression coverage.

File: scripts/development-workflow/workflow-lib.sh [hunk 2]
```diff
@@ -2642,13 +2685,16 @@ list_open_workflow_type_issues() {
     return 0
   fi
 
-  if ! printf '%s' "$project_items" | jq --argjson open "$open_issues" '
+  if ! printf '%s' "$project_items" | jq --argjson open "$open_issues" --arg type_field_name "$type_field_name" '
+    def classification_type:
+      (.[$type_field_name] // .type // .customType // ."Custom Type" // ."CustomType" // .customtype // "");
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply scripts/development-workflow/workflow-lib.sh]: Fixed in 308581c by making Workflow discovery prefer the configured/custom classification value over legacy `.type` when `type_field` is configured. Added `workflow_type_discovery_custom_type_beats_stale_type` regression coverage.

**Comment by lhpaul** [comment]
bugbot run

**Comment by lhpaul** [comment]
Advisory disposition: accepted as non-blocking. The duplicate PR-Agent Possible Issue points at the targeted-read fallback block, but its premise is incorrect: the `type` GraphQL alias is bound to the configured `$typeFieldName`, so configured `CustomType` / `Custom Type` is evaluated before fallback aliases. Bugbot found the actual precedence issues, and commit 308581c fixed them with regression coverage.
